pyteal/ast/router.py

```diff
@@ -831,9 +844,10 @@ def add_method_handler(
         """
         if not isinstance(method_call, ABIReturnSubroutine):
             raise TealInputError(
-                "for adding method handler, must be ABIReturnSubroutine"
+                "for adding method handler, must be ABIReturnSubroutine but method_call is {type(method_call)}"
             )
         method_signature = method_call.method_signature(overriding_name)
+        final_name = overriding_name or method_call.name()
```

Comment:
using only name makes me a bit worried, for `overriding_name` can be used to share same of different method:
`add(uint32 a, uint32b)uint32` and `add(uint64 a, uint64 b)uint64`. So I would propose to use `final_signature` here.

Reply:
cf. [da6c2de](https://github.com/algorand/pyteal/pull/634/commits/da6c2de4568c16f47716bd07ed82e0b841a2cee7)